winpr/libwinpr/path/path.c

```diff
@@ -634,27 +634,29 @@ HRESULT PathCchStripToRootW(PWSTR pszPath, size_t cchPath)
 HRESULT PathCchStripPrefixA(PSTR pszPath, size_t cchPath)
 {
 	BOOL hasPrefix;
-	BOOL deviceNamespace;
 
 	if (!pszPath)
-		return S_FALSE;
+		return E_INVALIDARG;
 
-	if (cchPath < 4)
-		return S_FALSE;
+	if (cchPath < 4 || cchPath > PATHCCH_MAX_CCH)
+		return E_INVALIDARG;
 
 	hasPrefix = ((pszPath[0] == '\\') && (pszPath[1] == '\\') &&
 		(pszPath[2] == '?') && (pszPath[3] == '\\')) ? TRUE : FALSE;
 
 	if (hasPrefix)
 	{
-		if (cchPath < 7)
+		if (cchPath < 6)
 			return S_FALSE;
```

Comment:
Yes, meaning it doesn't have the prefix. Windows also returns S_FALSE in this case